parsers/jscript.c

```diff
@@ -2908,6 +2908,13 @@ static bool parseStatementRHS (tokenInfo *const name, tokenInfo *const token, st
 				state->indexForName = makeFunctionTag (name, sig, false);
 			else
 				convertToFunction (state->indexForName, vStringValue (sig));
+
+			readToken (token);
+			if (isType (token, TOKEN_OPEN_CURLY))
+				parseBlock (token, state->indexForName);
+
+			if (isType (token, TOKEN_CLOSE_CURLY))
+				state->isTerminated = false;
```

Comment:
This is incorrect: `parseBlock()` will leave `token` on the *block's* close curly brace, which has nothing to do with the current statement's end.  See review